### PR TITLE
GH-481 Add feature to support center-aligned text

### LINKED ISSUE #481: Ability to centerize text.
The ability to centerize text with this syntax: `[center]text[/center]`

### PR DESCRIPTION
**Pull request by vLuckyyy**
The **LegacyColorProcessor** and **Legacy** classes were moved to a different package called 'adventure.legacy'. A new feature for center-aligned text was added using the **CenterTagResolver** class. To support this, **pixel-width** libraries were included in the build file.

Very thanks for @rchomczyk for psychical support.

https://github.com/EternalCodeTeam/EternalCore/assets/65517973/a9e3d4ae-1bf5-4cf1-830f-877e1e6cf260

Fixes #481

### CODE REVIEW

**Review by imDMK — APPROVED** [review]

**Review by rchomczyk — APPROVED** [review]

File: eternalcore-core/src/main/java/com/eternalcode/core/EternalCore.java [hunk 1]
```diff
@@ -240,6 +242,10 @@ public EternalCore(Plugin plugin) {
         this.audiencesProvider = BukkitAudiences.create(plugin);
         this.miniMessage = MiniMessage.builder()
                 .postProcessor(new LegacyColorProcessor())
+                .tags(TagResolver.builder()
+                        .resolver(CenterTagResolver.RESOLVER)
+                        .resolver(TagResolver.standard())
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/EternalCore.java:247]: To dodaje standardowe flagi czy setuje? (pytam defakto o metodę `.flags()`) bo jak dodaje to chyba nie trzeba dodawać standard

File: eternalcore-core/src/main/java/com/eternalcode/core/adventure/resolver/CenterTagResolver.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+public class CenterTagResolver implements Modifying {
+
+    public static final String TAG_NAME = "center";
+
+    public static final TagResolver RESOLVER = TagResolver.resolver(TAG_NAME, new CenterTagResolver());
+
+    @Override
+    public Component apply(@NotNull Component current, int depth) {
+        if (current instanceof TextComponent component && !((TextComponent) current).content().isBlank()) {
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/adventure/resolver/CenterTagResolver.java:18]:
```suggestion
        if (current instanceof TextComponent component && !component.content().isBlank() {
```

File: eternalcore-core/src/main/java/com/eternalcode/core/adventure/resolver/CenterTagResolver.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            return CenterAPI.center(component);
+        }
+
+        return Component.empty().mergeStyle(current);
```
Comment (Rollczi) [inline eternalcore-core/src/main/java/com/eternalcode/core/adventure/resolver/CenterTagResolver.java:22]: Nie wystarczy 
```suggestion
        return current;
```
?

Comment (vLuckyyy) [reply eternalcore-core/src/main/java/com/eternalcode/core/adventure/resolver/CenterTagResolver.java:22]: This is not working, if I try to use this suggestion, a center text is duplicated.

**Review by Rollczi — APPROVED** [review]

**Review by Jakubk15 — APPROVED** [review]

**Pull request merged**